### PR TITLE
Use locally installed `oc` if possible

### PR DESCRIPTION
**Pull request by ifireball**
Determine if the right version of the `oc` binary is installed in `$PATH` and use it if it does.

### CODE REVIEW

**Comment by arilivigni** [comment]
We still want to download the client in the default state.   That's why this was put here since people complained that they have to download the client themselves.   Did you put a conditional around it?  We're not going to remove that logic.

**Comment by ifireball** [comment]
Local `oc` is only used if found on `$PATH` and matches the version requested in thew configuration - otherwise the previous downloading logic runs.

**Comment by ifireball** [comment]
Looks like I have a bug, will fix

File: playbooks/roles/os_temps/tasks/login_to_cluster.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 # login to the cluster
 - name: "Login to the Openshift cluster"
-  shell: "{{ oc_bin }} login {{ cluster_ip }} --username={{ username }} --password={{ password }} --insecure-skip-tls-verify=true"
\ No newline at end of file
+  command: "{{ oc_bin }} login {{ cluster_ip|quote }} --username={{ username|quote }} --password={{ password|quote }} --insecure-skip-tls-verify=true"
```
Comment (arilivigni) [inline playbooks/roles/os_temps/tasks/login_to_cluster.yml]: Will this login maintain within the shell if done this way for the next call?

Comment (ifireball) [reply playbooks/roles/os_temps/tasks/login_to_cluster.yml]: yes. its seems its stored in a file somewhere, so it does not even matter which binary you've used to login.
Also seems to work across `oc` versions - but that is probably not guaranteed.

**Comment by arilivigni** [comment]
+1 to @dirgim that was similar to my comment just override and skip the install of the oc client

**Comment by ifireball** [comment]
@dirgim @arilivigni I would like to try and avoid adding yet another configuration variable given that we already have `oc_bin`.

How about if I check for existence, executability and version of the path `oc_bin` point to - and skip downloading if the test succeeds?

This way:
1. I can set it to simply `oc` on my system to get the behaviour I want
2. It will actually avoid downloading the 2nd time someone runs the script while still being able to download the 1st time

There is one thing Ill need to address which is what happens if the binary is in place but of the wrong version - I think I'll just download to a different location in that case (I wouldn't want to just blindly override it)

**Comment by arilivigni** [comment]
@ifireball I don't see the big deal if you have your options defined in a file adding one more does not mean you need it on the command line.  I think we are over complicating things here for a simple option override that still give you what you want.  I would be a -1 on this change.

**Comment by arilivigni** [comment]
@ifireball an even easier solution is to symlink your version into the normal location to minishift and we can put a check to see if the version is there and then not download it.  This avoids an extra option and avoids the time to download

**Comment by ifireball** [comment]
@arilivigni just to make sure I fully understand what you're saying - you opposed to any form of automated checking for wither `oc` is in place, and instead prefer if we just have a configuration options where we say that.

**Comment by arilivigni** [comment]
@ifireball I think it is fine to check if it is downloaded and not download the same version again.  In your case a symlink would suffice.  This way you download it once for the version that needs to be used.  Server and client should match.

**Pull request closed without merging**